### PR TITLE
feat(image_cache): add dump info ability for image_cache

### PR DESCRIPTION
**Pull request by W-Mai**
### Description of the feature or fix

There are two patches here. The first is to add `iter` ability to lvgl, through which the interface of any `iterable` object can be `unified`.

The second is the image cache info dump capability based on this `iter` ability, which is convenient for debugging the cache information that already exists in the image cache.

```plaintext
[User]	(1.083, +0)	 lv_image_cache_dump: Image cache dump: lv_image_cache.c:113
[User]	(1.083, +0)	 lv_image_cache_dump: 	size	data_size	cf	rc	type	decoded			src lv_image_cache.c:114
[User]	(1.083, +0)	 iter_inspect_cb: 	300x300    360000	16	1	file	0x103b71be8 	/res/img_1.png lv_image_cache.c:174
[User]	(1.083, +0)	 iter_inspect_cb: 	285x285    324900	16	10	file	0x103b88b18 	/res/img_0.bin lv_image_cache.c:174
```

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by FASTSHIFT** [comment]
Please add cache documentation.

**Comment by kisvegabor** [comment]
I've just opened https://github.com/lvgl/lvgl/issues/6671 to track the missing docs. 

Adding docs for the cache would be really great. It doesn need to explain RB tree or so, however it should introduce the API, show some basic examples, and tell a few words about image a freetype cache.

File: src/misc/lv_iter.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ **********************/
+
+typedef lv_result_t (*lv_iter_next_cb)(void * instance, void * context, void * elem);
+typedef void (*lv_iter_inspect_cb)(void * elem);
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+lv_iter_t * lv_iter_create(void * instance, uint32_t elem_size, uint32_t context_size, lv_iter_next_cb next_cb);
```
Comment (kisvegabor) [inline src/misc/lv_iter.h:45]: Add API docs.

File: src/misc/cache/lv_cache.h [hunk 2]
```diff
@@ -215,6 +215,8 @@ void lv_cache_set_name(lv_cache_t * cache, const char * name);
  */
 const char * lv_cache_get_name(lv_cache_t * cache);
 
+lv_iter_t * lv_cache_iter_create(lv_cache_t * cache);
```
Comment (kisvegabor) [inline src/misc/cache/lv_cache.h:223]: Add API docs

**Comment by W-Mai** [comment]
Update the API docs

File: src/misc/cache/lv_cache.h (hunk 2 above)
Comment (kisvegabor) [reply src/misc/cache/lv_cache.h:223]: Still no API docs here :stuck_out_tongue:

Comment (W-Mai) [reply src/misc/cache/lv_cache.h:223]: Oh sorry, I missed this 😄

**Review by kisvegabor — APPROVED** [review]
Thank you! :slightly_smiling_face: 

I know that you have high workload now, but let's document caching in the near future.

**Review by FASTSHIFT — APPROVED** [review]

**Comment by W-Mai** [comment]
> Thank you! 🙂
> 
> I know that you have high workload now, but let's document caching in the near future.

Sure😭, I will update the related documents in the near future.

**Pull request merged**